models/schemas/notificationSchema.ts

```diff
@@ -16,12 +16,12 @@ const NotificationSchema = new mongoose.Schema<Notification>({
   source: {
     type: mongoose.Schema.Types.ObjectId,
     refPath: 'sourceModel',
-    required: true,
+    required: false,
```

Comment:
Why change this to false?

Reply:
Is the idea that an achievement notification would have no source? Hmm... I guess that makes sense

Reply:
yea sspenst recommended this change for that reason. Leave as is?